src/module/system/damage/weapon.ts

```diff
@@ -27,16 +27,24 @@ import { DegreeOfSuccessIndex, DEGREE_OF_SUCCESS } from "@system/degree-of-succe
 import { mapValues, objectHasKey, sluggify } from "@util";
 import { createDamageFormula, AssembledFormula } from "./formula";
 import { nextDamageDieSize } from "./helpers";
-import { DamageDieSize, DamageFormulaData, MaterialDamageEffect, WeaponDamageTemplate } from "./types";
+import {
+    DamageDieSize,
+    DamageFormulaData,
+    DamageRollContext,
+    MaterialDamageEffect,
+    WeaponDamageTemplate,
+} from "./types";
+import { DamageModifierDialog } from "./modifier-dialog";
 
 class WeaponDamagePF2e {
-    static calculateStrikeNPC(
+    static async calculateStrikeNPC(
         attack: MeleePF2e,
         actor: NPCPF2e | HazardPF2e,
         actionTraits: TraitViewData[] = [],
         proficiencyRank = 0,
-        options: Set<string> = new Set()
-    ): WeaponDamageTemplate | null {
+        options: Set<string> = new Set(),
+        context: DamageRollContext
+    ): Promise<WeaponDamageTemplate | null> {
```

Comment:
If you're adding new params, might as well convert it to an object like WeaponDamagePF2e#calculate(). That said, isn't options part of context?

Reply:
I'm not sure what the intention was there anymore. The options are indeed part of the context.